packages/core/src/exchange/acceptOffer.ts

```diff
@@ -49,26 +47,20 @@ const approveTransferTransaction = async (
   signer: Signer,
 ): Promise<TransactionResponse | null> => {
   invariant(offer)
-  if (offer.type === 'BUY')
-    return approveCollection(
-      sdk,
-      {
-        chain: offer.asset.collection.chainId,
-        collection: toAddress(offer.asset.collection.address),
-      },
-      signer,
-    )
-  // The offer is in native token so we don't need any approval
-  if (!offer.currency.address) return null
-  return approveCurrency(
-    sdk,
-    {
-      chain: offer.currency.chainId,
-      currency: toAddress(offer.currency.address),
-      amount: BigNumber.from(offer.unitPrice).mul(quantity).toString(),
-    },
-    signer,
-  )
+
+  const address = await signer.getAddress()
+
+  const {
+    createOfferApprovalTransaction: { transaction },
+  } = await sdk.CreateOfferApprovalTransaction({
+    offerId: offer.id,
+    accountAddress: toAddress(address),
+    quantity: quantity.toString(),
+  })
+
+  if (!transaction) return null
```

Comment:
I think this one should stay as it is and be left untouched for compatibility purpose especially with the lazymint that is not supported by this new approval